### PR TITLE
Doc: Adding ftrace documentation

### PR DESCRIPTION
**Pull request by jbech-linaro**
Adding a section in the debug.md file that explains how to enable ftrace
and use it for debugging the kernel module for OP-TEE.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>
Tested-by: Joakim Bech <joakim.bech@linaro.org> (QEMU, Hikey)

### CODE REVIEW

File: documentation/debug.md [hunk 1]
```diff
@@ -141,10 +142,163 @@ gvd --debugger $HOME/devel/toolchains/gcc-linaro-arm-none-eabi-4.9-2014.09_linux
...
+echo 0 > options/graph-time
+```
+
+# 3. Known issues
 1. Printing the call stack using `bt` makes gdb go into an endless loop.
    Temporary workaround, in gdb, instead of simply writing `bt`, also mention
    how many frames you would like to see, for example `bt 10`.
 2. Cannot set breakpoints when the system is up and running. Workaround, set the
-   breakpoints before booting up the system.
+   breakpoints befora booting up the system.
```
Comment (jbech-linaro) [inline documentation/debug.md]: I've spotted this error, will revert it when applying tags or addressing other comments.

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jbech-linaro** [comment]
Spelling error fixed and tags applied.

**Comment by pascal-brand-st-dev** [comment]
@jbech-linaro Can you rebase it on top of master?

**Comment by jbech-linaro** [comment]
Absolutely, give me a second or two and I'll send a rebased patch.

**Comment by jbech-linaro** [comment]
Rebased patch has been submitted.

**Pull request merged**